libqtile/scripts/migrate.py

```diff
@@ -115,6 +119,54 @@ def new_at_current_to_new_client_position(config):
     )
 
 
+def textbox_widget_padding():
+    """
+    Adding padding mixin to _TextBox means configs need to
+    be updated. "padding" should just impact left/right padding
+    so needs to be replaced with "padding_x" to allow vertical
+    padding to be adjusted independently.
+    """
+    def update_padding(ln, cap, fn):
+        for c in cap.get("class_arguments"):
+            for leaf in c.leaves():
+                if leaf.value == "padding":
+                    leaf.value = "padding_x"
+
+    migrations = []
+
+    textwidgets = [getattr(libqtile.widget, w) for w in libqtile.widget.__all__]
+    textwidgets = [widget.__name__ for widget in textwidgets if inspect.isclass(widget) and issubclass(widget, _TextBox)]
+
+    for widget in textwidgets:
+
+        # Custom selector to match widget.WidgetName and WidgetName instances
+        selector = r"""
+            class_call=power<
+                class_name='{widget}'
+                trailer< '(' class_arguments=any* ')' >
+            >
+            |
+            class_call=power<
+                'widget'
+                trailer< '.' '{widget}' >*
+                trailer< '(' class_arguments=any* ')' >
+                any* 
+            >
+        """.format(widget=widget)
```

Comment:
There's probably a neater way of doing this selector but I'm very new to `bowler`!

Reply:
We all are :)